src/client/scripts/room.js

```diff
@@ -1,41 +1,44 @@
 "use strict";
 /* global Cards:false Rooms:false*/
 /**
-*@author THouse
 *@purpose To provide the room template with data to display
 **/
 
 Template.room.onCreated(function () {
-    this.subscribe("cards");
+    Meteor.autorun(function() {
+        Meteor.subscribe("cards", Session.get("roomNumber"));
+    });
 });
 
 Template.room.helpers({
     categories: function() {
-        return Rooms.findOne(
-            {"roomCode": Session.get("roomNumber")}
-        ).categories;
+        var room = Rooms.findOne({"roomCode" : Session.get("roomNumber")});
+
+        return room.categories;
     },
 
+    //TODO have this call another mentod
```

Comment:
if you put a sort field in there it fixes the ordering issue.
e.g. Cards.find({query selectors},{$sort:{createdAt:-1});